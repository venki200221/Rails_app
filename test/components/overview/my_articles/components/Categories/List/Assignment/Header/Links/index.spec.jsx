import React from 'react';
import { shallow, mount } from 'enzyme';
import '../../../../../../../../../testHelper';

import MyArticlesAssignmentLinks from '@components/overview/my_articles/components/Categories/List/Assignment/Header/MyArticlesAssignmentLinks.jsx';

describe('MyArticlesAssignmentLinks', () => {
  const props = {
    articleTitle: 'title',
    assignment: { id: 1, article_url: 'url', role: 0 },
    courseType: 'ClassroomProgramCourse',
    current_user: { id: 99, username: 'user' },
    project: 'wikipedia'
  };

  it('should show the default links', () => {
    const component = shallow(<MyArticlesAssignmentLinks {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should not show the bibliography link if the course type is anything but ClassroomProgramCourse', () => {
    const component = mount(<MyArticlesAssignmentLinks {...props} courseType="Editathon" />);
    expect(component).toMatchSnapshot();
    expect(component.find('BibliographyLink').length).toBeUndefined;
  });

  it('should show the peer review link if the assignment role is set to reviewing', () => {
    const newProps = { ...props, assignment: { ...props.assignment, role: 1 } };
    const component = mount(<MyArticlesAssignmentLinks {...newProps} />);
    expect(component.find('PeerReviewLink').length).toEqual(1);
  });

  it('should show GroupMembersLink and ReviewerLink if there are editors or reviewers', () => {
    const assignment = {
      ...props.assignment,
      editors: ['editor'],
      reviewers: ['reviewer']
    };
    const component = mount(<MyArticlesAssignmentLinks {...props} assignment={assignment} />);
    expect(component.find('GroupMembersLink').length).toEqual(1);
    expect(component.find('AllPeerReviewLinks').length).toEqual(1);
  });
});
