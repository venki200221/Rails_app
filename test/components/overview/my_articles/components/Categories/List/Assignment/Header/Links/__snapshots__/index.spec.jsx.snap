// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyArticlesAssignmentLinks should not show the bibliography link if the course type is anything but ClassroomProgramCourse 1`] = `ReactWrapper {}`;

exports[`MyArticlesAssignmentLinks should show the default links 1`] = `ShallowWrapper {}`;
